File: src/main/java/com/github/hibi_10000/plugins/plategate/command/PGList.java
package com.github.hibi_10000.plugins.plategate.command;

import com.github.hibi_10000.plugins.plategate.PlateGate;
import com.github.hibi_10000.plugins.plategate.util.Util;
import net.md_5.bungee.api.chat.ClickEvent;
import net.md_5.bungee.api.chat.HoverEvent;
import net.md_5.bungee.api.chat.TextComponent;
import net.md_5.bungee.api.chat.hover.content.Text;
import org.bukkit.Bukkit;
import org.bukkit.OfflinePlayer;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.List;

public class PGList {
	
	private final PlateGate plugin;
	private final Util util;
	public PGList(PlateGate instance) {
		this.plugin = instance;
		this.util = new Util(instance);
	}
	
	
	public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {

		if (!(sender.hasPermission("plategate.list"))) {
			sender.sendMessage("§a[PlateGate] §c権限が不足しています。");
			return false;
		}

		if (!(args.length == 2 || args.length == 1)) {
			TextComponent help = new TextComponent("§a[PlateGate] §cコマンドが間違っています。 /" + label +" help で使用法を確認してください。");
			help.setHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new Text("§aクリックで§b\"/" + label + " help\"§aを実行")));
			help.setClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/" + label + " help"));
			sender.spigot().sendMessage(help);
			return false;
		}

		Player searchp;
		if (sender.hasPermission("plategate.admin") && args.length == 2) {

			boolean args1player = false;
			for (Player p : Bukkit.getOnlinePlayers()) {
				if (p.getName().equalsIgnoreCase(args[1])) {
					args1player = true;
					break;
				}
			}
			for (OfflinePlayer p : Bukkit.getOfflinePlayers()) {
				if (p.getName().equalsIgnoreCase(args[1])) {
					args1player = true;
					break;
				}
			}
			if (!args1player) {
				sender.sendMessage("§a[PlateGate] §cそのプレイヤーは存在しません。");
				return false;
			}

			searchp = Bukkit.getPlayer(args[1]);
		} else {
			searchp = (Player) sender;
		}

		//List<JsonObject> jolist = new JsonHandler(plugin).JsonRead(searchp, null);

		for (String index : util.IndexJson("owner", searchp.getUniqueId().toString(), (Player) sender)) {
			sender.sendMessage("§a[PlateGate] §bPlayer §6" + searchp.getName() + " §bが所有しているGate一覧");
			if (util.getJson(index, "to", (Player) sender).equalsIgnoreCase("")) {
				sender.sendMessage(" §b" + util.getJson(index, "name", (Player) sender));
			//} else if (new JsonHandler(plugin).JsonRead(jog.get("to").getAsString(), null).getAsJsonObject()
			//		.get("to").getAsString() == jog.get("name").getAsString()) {
			//	sender.sendMessage(" " + jog.get("name").getAsString() + " <--> " + jog.get("to").getAsString());
			} else {
				sender.sendMessage(" §b" + util.getJson(index, "name", (Player) sender) + " §a---> §b" + util.getJson(index, "to", (Player) sender));
			}
		}


		return true;
	}
	
	public List<String> onTabComplete(CommandSender sender, Command command, String alias, String[] args) {
		
		return null;
	}
}

File: src/main/kotlin/com/github/hibi_10000/plugins/plategate/command/PGList.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package com.github.hibi_10000.plugins.plategate.command

import com.github.hibi_10000.plugins.plategate.MessageUtil
import com.github.hibi_10000.plugins.plategate.Util
import com.github.hibi_10000.plugins.plategate.dbUtil
import com.github.hibi_10000.plugins.plategate.localization.Message
import net.md_5.bungee.api.chat.TextComponent
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

object PGList {
    @Suppress("UNUSED_PARAMETER")
    fun onCommand(sender: Player, command: Command, label: String, args: Array<String>): Boolean {
        if (!Util.checkPermission(sender, "plategate.command.list")) return false
        if (args.size != 1) return Util.commandInvalid(sender, label)

        val senderInfo = MessageUtil.getSenderInfo(sender)
        senderInfo.text = " §6${senderInfo.text} "
        MessageUtil.send(sender, Message.COMMAND_LIST_HEADER, senderInfo)
        val gateList = try {
            dbUtil.getList(sender.uniqueId)
        } catch (e: Exception) {
            MessageUtil.catchUnexpectedError(sender, e)
            return false
        }
        for (gate in gateList) {
            val gateInfo = MessageUtil.getGateInfo(gate, sender)
            gateInfo.text = " §b${gateInfo.text} "
            if (gate.toOwner == null || gate.toName == null) {
                sender.spigot().sendMessage(gateInfo)
            } else {
                val toGate = try {
                    dbUtil.get(gate.toOwner!!, gate.toName!!)
                } catch (e: Exception) {
                    MessageUtil.catchUnexpectedError(sender, e)
                    sender.spigot().sendMessage(gateInfo)
                    continue
                }
                val arrow = TextComponent("§a--->")
                if (toGate.toName == gate.name && toGate.toOwner == gate.owner) {
                    arrow.text = "§a<-->"
                }
                val toGateInfo = MessageUtil.getGateInfo(toGate, sender)
                toGateInfo.text = " §b${toGateInfo.text} "
                if (gate.toOwner != sender.uniqueId) {
                    val toOwner = Util.getPlayer(gate.toOwner!!, sender)
                    toGateInfo.text += "(Owner: ${toOwner?.name})"
                }
                sender.spigot().sendMessage(gateInfo, arrow, toGateInfo)
            }
        }
        return true
    }

    @Suppress("UNUSED_PARAMETER")
    fun onTabComplete(sender: CommandSender, command: Command, alias: String, args: Array<String>): List<String>? {
        return null
    }
}
